crates/core/tedge_agent/src/tedge_to_te_converter/converter.rs

```diff
@@ -19,20 +23,56 @@ impl Converter for TedgetoTeConverter {
 }
 
 impl TedgetoTeConverter {
+    /// Creates a new converter with default prefix `"te"`.
+    #[cfg(test)]
     pub fn new() -> Self {
-        TedgetoTeConverter {}
+        Self {
+            mqtt_schema: MqttSchema::new(),
+        }
+    }
+
+    pub fn with_root(root: String) -> Self {
+        Self {
+            mqtt_schema: MqttSchema::with_root(root),
+        }
     }
 
     fn try_convert(&mut self, message: MqttMessage) -> Vec<tedge_mqtt_ext::Message> {
-        match message.topic.clone() {
+        match &message.topic {
             topic if topic.name.starts_with("tedge/measurements") => {
                 self.convert_measurement(message)
             }
             topic if topic.name.starts_with("tedge/events") => self.convert_event(message),
             topic if topic.name.starts_with("tedge/alarms") => self.convert_alarm(message),
-            topic if topic.name.starts_with("tedge/health") => {
+
+            // to be able to move different services to new health topic at different times, we
+            // selectively map services either from old to new topics, and vice versa.
+            // tedge-agent publishes on new health topic, exclude it from old mapping not to produce
+            // a publish loop
+            topic
+                if topic.name.starts_with("tedge/health")
+                    && !topic.name.contains("tedge-agent") =>
+            {
                 self.convert_health_status_message(message)
             }
```

Comment:
Would not be simpler to update all the services in a single PR?

Or even simpler:

- to start with the service *consuming* health check messages (the mapper and [tedge-watchdog](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_watchdog/src/systemd_watchdog.rs#L89), so all the health status are consumed from the `te` topics.
- to update one after the other the services (so they publish health status on the `te` topics).

I realized for that to work, one needs to update the `tedge_to_te_converter` to also converts health check requests. So, we will be able to update the services (as the agent) one after the other so they react to `te/+/+/+/+/cmd/health/check`.

The very last step will be to update tedge-watchdog to send health check requests on the  `te` topics.

Reply:
You're right, it would be much simpler and it's something that we have to do anyway, so better to do it now.